src/main/kotlin/app/revanced/patcher/extensions/Extensions.kt

```diff
@@ -177,6 +180,21 @@ fun MutableMethod.addInstructions(index: Int, smali: String, externalLabels: Lis
             // this line works because we created the dummy instructions in the same order as the externalLabels list.
             val (_, instruction) = externalLabels[(compiledInstructions.size - 1) - labelIndex]
             instruction.makeNewLabel()
+
+            // Since we delete the nops as we're iterating, to find the index of subsequent
+            // nops we need to substract the number of nops already removed
+            val nopInstructionIndex = index + labelIndex - nopInstructionsRemoved;
+
+            // Remove the corresponding nop.
+            if (nopInstructionsRemoved < externalLabels.size)
```

Comment:
Instead of this, why not just add the compiled instructions without the nops to the target method implementation instead of adding the entire instruction list, and then removing the nop instructions from them? Above the instructions are compiled, you could instead just add a subList of the instructions without the nops.

Reply:
Something like this?

Reply:
I think so, is this tested?

Reply:
Yeah it works, but I think it looks a little awkward.